File: 429/B/main.py
# 自分の作ったコード O(N^2)

N, M = map(int, input().split())

A = list(map(int, input().split()))

count = 0
no_count = 0

for i in range(len(A)):
    B = A.copy()

    # i番目の要素を削除する
    B.pop(i)

    # Bの合計を求める
    total = 0
    for x in B:
        total += x
    
    if total == M:
        count += 1
    else:
        no_count += 1

if count > 0:
    print("Yes")
else:
    print("No")

# もっと速いコード
# n, m = map(int, input().split())
# a = list(map(int, input().split()))

# if sum(a) - m in a:
#     print("Yes")
# else:
#     print("No")

File: 429/B/main.go
package main

import "fmt"

func sum(arr []int) int {
    total := 0
    for _, v := range arr {
        total += v
    }
    return total
}

func main() {
    var N, M int
    fmt.Scan(&N, &M)

    A := make([]int, N)
    for i := 0; i < N; i++ {
        fmt.Scan(&A[i])
    }

    // 合計を出す
    result := sum(A)

    for _, v := range A {
        if result - M == v {
            fmt.Println("Yes")
            return
        }
    }
    fmt.Println("No")
}